core/src/types/blocking_read/blocking_reader.rs

```diff
@@ -29,7 +31,9 @@ use crate::*;
 /// BlockingReader is designed to read data from given path in an blocking
 /// manner.
 pub struct BlockingReader {
-    pub(crate) inner: oio::BlockingReader,
+    // pub(crate) inner: oio::BlockingReader,
+    pub(crate) iter: BufferIterator,
```

Comment:
Hi, please re-visit the impl of `Reader`. `oio::BlockingReader` can be changed into `Arc<dyn oio::BlockRead>`.

So:

- We can build a new `BufferIterator` everytime we call `read` or other API.
- The `BufferIterator` is used internally.

Reply:
Plz review the code again.

Reply:
@Xuanwo Why memory-leak test failed, Do you have any idea?